### PR TITLE
Sink values and end to end test

### PR DESCRIPTION
**Pull request by dcolligan**
change cli sink values from empty string to None

add end to end test

Questions: 
1) do want want to incorporate this new end-to-end test into the travis CI run?  If so, we have one of two choices: a) make a dummy backend for the ga4gh_server that doesn't require any genome data to be on the file system b) make sure data is on the file system when Travis runs somehow
2) what test suite should the test I've written be in?  or...
3) should we factor out the more general functionality I've written into a test framework for one of the suites?

### CODE REVIEW

File: ga4gh/cli.py [hunk 1]
```diff
@@ -196,6 +202,9 @@ def addGlobalOptions(parser):
     parser.add_argument(
         "--config-file", "-F", type=str,
         help="The configuration file to use")
+    parser.add_argument(
```
Comment (jeromekelleher) [inline ga4gh/cli.py]: Why do we want this? The reloader is generally pretty handy, and this server is just for development...

**Comment by jeromekelleher** [comment]
I'm still a bit hazy on why `default=None` is better than `default=''`. Sorry for being slow, but could you explain this again?

I like the end-to-end test, but will need a bit of time to digest what's going on here.

File: ga4gh/cli.py (hunk 1 above)
Comment (dcolligan) [inline ga4gh/cli.py]: When starting the server within a test, we want one instance of the server to be launched, not two processes.  If the later, it makes it difficult to kill the server.

(Reloading is only useful to pick up changes in code when developing; when running a test this should not be happening.)

**Comment by dcolligan** [comment]
Because `default=''` also requires special-case handling code, and is a less obvious sink value.  The likely result is that we send requests to the server that erroneously have `attribute: ''` instead of what we really want, `attribute: None`

**Comment by dcolligan** [comment]
I've rearranged the code a bit and the Travis CI build is now failing in a way that I expected it to (because the data is not there) and reporting the error as such.

**Comment by jeromekelleher** [comment]
I think we should defer this until after we've got simulated data working --- it's too difficult to do any real testing otherwise.

**Comment by dcolligan** [comment]
Agreed, this change depends on a simulated backend.

**Comment by dcolligan** [comment]
Closing for now

**Pull request closed without merging**